stores/submissionStore.ts

```diff
@@ -1,57 +1,45 @@
 import { gql } from 'graphql-request'
 import { defineStore } from 'pinia'
-import { ref, type Ref } from 'vue'
-import { gqlClient } from '../utils/graphql.js'
-import type { Locale, CreateSubmissionInput, Submission } from '~/typedefs/gqlTypes'
+import { type Ref, ref } from 'vue'
+import { gqlClient, graphQLClientRequestWithRetry } from '../utils/graphql.js'
+import type { ServerResponse, ErrorCode, ServerErrorResponse } from '~/typedefs/serverResponse.js'
+import type { MutationCreateSubmissionArgs, Submission } from '~/typedefs/gqlTypes'
+
+export interface SubmissionFormInputFields {
+    location: string
+}
 
 export const useSubmissionStore = defineStore('submissionStore', () => {
-    const location: Ref = ref('')
-    const firstName: Ref = ref('')
-    const lastName: Ref = ref('')
-    const selectLanguage1: Ref = ref('')
-    const selectLanguage2: Ref = ref('')
-    const otherNotes: Ref = ref('')
     const submissionCompleted: Ref = ref(false)
 
-    async function submit() {
-        const spokenLanguages: Locale[] = []
-
-        if (selectLanguage1.value !== '') {
-            spokenLanguages.push(selectLanguage1.value as Locale)
-        }
-
-        if (selectLanguage2.value !== '') {
-            spokenLanguages.push(selectLanguage2.value as Locale)
-        }
-
-        const submission = {
-            googleMapsUrl: location.value,
-            healthcareProfessionalName: `${firstName.value} ${lastName.value}`,
-            spokenLanguages: spokenLanguages,
-            notes: otherNotes.value
-        } satisfies CreateSubmissionInput
+    async function createNewSubmission(newSubmission: MutationCreateSubmissionArgs):
+    Promise<ServerResponse<Submission>> {
+        let response: ServerResponse<Submission> = { data: {} as Submission, errors: [], hasErrors: false }
 
         try {
-            await gqlClient.request<Submission>(createSubmissionMutation, {
-                input: submission
-            })
-            submissionCompleted.value = true
+            response = await graphQLClientRequestWithRetry(
+                gqlClient.request.bind(gqlClient),
+                createSubmissionMutation,
+                newSubmission
+            ) as ServerResponse<Submission>
         } catch (e) {
             console.error(`There was an error creating the submission ${e}`)
+            const ServerErrorResponse = e as ServerErrorResponse
+
+            ServerErrorResponse.response.errors.forEach(errorResponse => {
```

Comment:
It would be more readable if you used a `map()` function and set the response.errors to the new map array. 
Tip:
any time you're changing the shape of an object or iterating through something that you are going to assign to another variable, use `map()`
`forEach()` is better for iterating and doing an action on each item. (like rendering or saving each one)